app/(playground)/p/[agentId]/canary/actions.ts

```diff
@@ -106,15 +108,22 @@ type ActionSource = TextSource | TextGenerationSource | FileSource;
 
 export async function action(
 	artifactId: ArtifactId,
-	graphUrl: string,
+	agentId: AgentId,
 	nodeId: NodeId,
 ) {
 	const lf = new Langfuse();
 	const trace = lf.trace({
 		sessionId: artifactId,
 	});
 
-	const graph = await fetch(graphUrl).then(
+	const agent = await db.query.agents.findFirst({
+		where: (agents, { eq }) => eq(agents.id, agentId),
+	});
+	if (agent === undefined || agent.graphUrl === null) {
```

Comment:
nits:

Wouldn't it be possible to specify `agent.graphUrl` in the query at line 120 instead of determining it here?

Reply:
As it is a Nullable column, the type definition will be Null even if it is narrowed down in a query.

This will be changed in the future as it will no longer be needed if it is set to Not null.

https://github.com/giselles-ai/giselle/blob/cce94aa476c30265fc62fdf5d3ae1f38a2d308e9/drizzle/schema.ts#L135-L136